src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -5051,6 +5051,8 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
         {
             assert(sig->numArgs == 2);
 
+            impSpillSideEffect(true, verCurrentState.esStackDepth - 2 DEBUGARG("Spilling op1 for ZeroHighBits"));
```

Comment:
Given that this is just a fixup to make the encoding easier; it might be better to not spill, to ensure that the operand order here is correct, and instead to just do the "fixup" in lowering where we know its safe to do.

Reply:
Isn't it what I meant by moving it to `gtSetEvalOrder` ? 🙂 (the gtSetEvalOrder, or its "swap operands" part itself could be moved to lower).

Reply:
Ah, I didn't realize you meant that as well; 👍 

That being said, I'm not sure that `gtSetEvalOrder` is the right thing here either. This isn't really about evaluation order at all, rather its purely about putting operands in the expected place for codegen because these instructions get encoded "backwards" from most instructions.

More ideally we'd just do it in codegen or in emit proper, with the former likely being less expensive.